tools/monorepo-utils/src/slack/commands/slack/slack-message.ts

```diff
@@ -0,0 +1,66 @@
+/**
+ * External dependencies
+ */
+import { Command } from '@commander-js/extra-typings';
+
+/**
+ * Internal dependencies
+ */
+import { Logger } from '../../../core/logger';
+import { requestAsync } from '../../../core/util';
+
+type SlackResponse = {
+	ok: boolean;
+	error?: string;
+};
+
+export const slackMessageCommand = new Command( 'message' )
+	.description( 'Send a plain-text message to a slack channel' )
+	.argument(
+		'<token>',
+		'Slack authentication token bearing required scopes.'
+	)
```

Comment:
What do you think about gathering this token from environment variables instead of the command line argument. We do this for `GITHUB_TOKEN` already. I think it would simplify development and CI will be gathering this value from environment variables anyways.

Reply:
@psealock I think you and I are at odds on this opinion. Environment variables can be passed to scripts so why make a script force you to create an env file or export a variable? args are always more flexible and if you want to use an env variable for some reason you can pass it to the command.

Reply:
Ok there is one situation you might pass a ENV variable, which is adjusting global behaviour of things like logging. I think thats ok, because we provide a consistent way to adjust logging behaviour across the whole CLI thats more convenient than args.

But args have their place, to document the API of the thing you're trying to do and for global behaviours like changing logging behaviour we can document them all in the top level help for the CLI

Reply:
>How does it simplify development or usage?

The next time I work on this command I'll need to dig the token from wherever I have it saved and copy/paste into the command line invocation. When its saved as environment variable, I won't need to deal with it after the first time.

Reply:
> The next time I work on this command I'll need to dig the token from wherever I have it saved and copy/paste into the command line invocation. When its saved as environment variable, I won't need to deal with it after the first time.

No offence meant, but I think thats weak reasoning for making the API of CLI tools less intuitive.

This is your one specific workflow. I personally use command history, or just recopy the token from my password manager. And then I never had to worry about accidentally leaving variables in my bash profile. Additionally, not relying on variables means I never accidentally leave one set that causes confusing behaviour. Who is right in this situation? neither of us, but my main argument that I think still holds is that your particular workflow is not just justification to make worse CLI tools.